src/core/tests/Test_StatusHandlerTruncation.py

```diff
@@ -0,0 +1,829 @@
+        self.__read_tmp_file_and_assert_log_msg(substatus_file_data, patch_count_assessment=patch_count)
+
+    def test_only_assessment_patches_over_size_limit_truncated(self):
+        """ Perform truncation on assessment patches.
+        Input: xxx assessment patches
+        Output:
+        operation: Assessment,
+        assessment substatus name: PatchAssessmentSummary,
+        assessment substatus status: warning,
+        assessment substatus truncated patches < patch_count,
+        assessment errors code: 2 (warning),
+        assessment errors details count: 0,
+        assessment message json fields == assessment truncated message json fields,
+        assert count of patches removed from log,
+        complete status file size > 128kb,
+        truncated status file size < 126kb < 128kb,
+        truncated status file size < completed status file size. """
+
+        self.runtime.execution_config.operation = Constants.ASSESSMENT
+        self.runtime.status_handler.set_current_operation(Constants.ASSESSMENT)
+
+        patch_count = random.randint(780, 1000)
+        test_packages, test_package_versions = self.__set_up_packages_func(patch_count)
+        self.runtime.status_handler.set_package_assessment_status(test_packages, test_package_versions)
+        self.runtime.status_handler.set_assessment_substatus_json(status=Constants.STATUS_SUCCESS)
+        self.runtime.status_handler.log_truncated_patches()
+
+        # Assert complete status file
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.complete_status_file_path, 'r') as file_handle:
+            complete_substatus_file_data = json.load(file_handle)
+
+        self.__assert_patch_summary_from_status(complete_substatus_file_data, Constants.ASSESSMENT, Constants.PATCH_ASSESSMENT_SUMMARY, Constants.STATUS_SUCCESS, patch_count)
+
+        # Assert assessment truncated status file
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.status_file_path, 'r') as file_handle:
+            truncated_substatus_file_data = json.load(file_handle)
+
+        self.__assert_patch_summary_from_status(truncated_substatus_file_data, Constants.ASSESSMENT, Constants.PATCH_ASSESSMENT_SUMMARY, Constants.STATUS_WARNING, patch_count, complete_substatus_file_data=complete_substatus_file_data, under_internal_size_limit=True, is_truncated=True)
+
+        # Assert all assessment fields in the message json are equal in both status files
+        self.__assert_assessment_truncated_msg_fields(complete_substatus_file_data, truncated_substatus_file_data)
+
+        # Assert 'Count of patches removed from: [Assessment=xxx] [Installation=0] log message is called
+        self.__read_tmp_file_and_assert_log_msg(truncated_substatus_file_data, patch_count_assessment=patch_count)
+
+    def test_only_assessment_patches_large_size_limit_truncated(self):
```

Comment:
What is the difference between this test and test_only_assessment_patches_over_size_limit_truncated()?

Reply:
when i was writing ut at first when the patch count was roughly > 1000 the ut was passing, but when i added a test with very large patch count I was getting timeout error then i know my code time complexity was bad so i had to use better approach to reduce the time complexity.  I thought it will be good to include two scenarios to easy identify code time performance issue. since you have concern of it, I've removed the lower count ut. 

done

Reply:
@feng-j678 I was merely asking for the difference between these 2 tests. Please be informed, any questions raised in PR feedback does not necessarily mean the approach should change, it can simply be reviewer trying to understand the line of thinking behind that piece of code

Reply:
oh i see. yeah, there's no difference between the 2 tests other than the input size. combining will be good since the current method description is very explicit, one will know why the ut failed if the time complexity has changed. 

done.